modules/core/src/main/scala/com/azavea/stac4s/extensions/asset/AssetCollectionExtension.scala

```diff
@@ -0,0 +1,23 @@
+package com.azavea.stac4s.extensions.asset
+
+import com.azavea.stac4s.extensions.CollectionExtension
+
+import io.circe._
+import io.circe.syntax._
+
+final case class AssetCollectionExtension(
```

Comment:
It turns out we'd implemented the `asset` extension? I don't know why we did that. But I moved it to the appropriate zone. We could consider moving implemented extensions into another module, or a module per extension or something? I don't think the second one is great because there are tons of extensions but the first one might be reasonable